### PR TITLE
Path Of Titans

### PR DESCRIPTION
**Pull request by Hendalf-DEV**
Path Of Titans server egg. Not tested properly but the server is running so I guess it works if you set all the required ports.

# Description

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [ ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:
  I'm adding not editing, so there is no need to have backup of anything.

* [ ] You verify that the start command applied does not use a shell script
  * [x] If some script is needed then it is part of a current yolk or a PR to add one
* [ ] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [ ] You verify that the start command applied does not use a shell script
    * [X] If some script is needed then it is part of a current yolk or a PR to add one
6. [ ] The egg was exported from the panel

### CODE REVIEW

**Comment by SulivanM** [comment]
Hello there,
I got an error installation : 
image

Best regards

**Comment by Hendalf-DEV** [comment]
> Hello there, I got an error installation : image
> 
> Best regards

Hello, sorry for long response, I am currentyly going to test this again, but I need time.

With kind regards, Hendalf

**Comment by gOOvER** [comment]
> Hello there, I got an error installation : image
> 
> Best regards

add this as variable or define it in the image

**Comment by Hendalf-DEV** [comment]
Hello again, I found the issue. It seems that POT does not want to run on Ubuntu image for some reason. So I reccomend change it to Debian image. That should work.

With kind regards, Hendalf.

p.s. I am going to make changes to this egg so no one will get to this trap again.

**Comment by Hendalf-DEV** [comment]
Providing additional images
image

**Comment by Hendalf-DEV** [comment]
The only thing that is additionally required is some of the ports
If there is possibility to make them accessable for the server without adding them additionally, it would be nice.

image

**Comment by gOOvER** [comment]
> The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> 
> image

you can only autoassign the primary port. For all others you must add variables

**Comment by Hendalf-DEV** [comment]
> > Hello again, I found the issue. It seems that POT does not want to run on Ubuntu image for some reason. So I reccomend change it to Debian image. That should work.
> > With kind regards, Hendalf.
> > p.s. I am going to make changes to this egg so no one will get to this trap again.
> 
> Somehow I can't believe that now? Do you have an error message why it doesn't work with Ubuntu?

Nope, it just gives the same thing as this person had, I changed to Debian and now it works.

**Comment by Hendalf-DEV** [comment]
> > The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> > image
> 
> you can only autoassign the primary port. For all others you must add variables

Atleast I got the query port assigned automatically. But I have no clue about how do I assign rcon port.

**Comment by gOOvER** [comment]
> > > The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> > > image
> > 
> > 
> > you can only autoassign the primary port. For all others you must add variables
> 
> Atleast I got the query port assigned automatically. But I have no clue about how do I assign rcon port.

there is only 1 variable which is autoassigned: SERVER_PORT.

For all others you must add a varibale, so i dont believe that Queryport was autoassigned, when there is no variable

**Comment by gOOvER** [comment]
> > > Hello again, I found the issue. It seems that POT does not want to run on Ubuntu image for some reason. So I reccomend change it to Debian image. That should work.
> > > With kind regards, Hendalf.
> > > p.s. I am going to make changes to this egg so no one will get to this trap again.
> > 
> > 
> > Somehow I can't believe that now? Do you have an error message why it doesn't work with Ubuntu?
> 
> Nope, it just gives the same thing as this person had, I changed to Debian and now it works.

thats not a problem with the Distro. Which image are you using now?

If steamcmd, then Dotnet is allready installed. In Ubuntu it isnt

**Comment by Hendalf-DEV** [comment]
> > > > The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> > > > image
> > > 
> > > 
> > > you can only autoassign the primary port. For all others you must add variables
> > 
> > 
> > Atleast I got the query port assigned automatically. But I have no clue about how do I assign rcon port.
> 
> there is only 1 variable which is autoassigned: SERVER_PORT.
> 
> For all others you must add a varibale, so i dont believe that Queryport was autoassigned, when there is no variable

I guess I made it unclear what I did mean by that. Asssigning screenshot.
I added server properties to the config file.

image

**Comment by gOOvER** [comment]
> > > > > The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> > > > > image
> > > > 
> > > > 
> > > > you can only autoassign the primary port. For all others you must add variables
> > > 
> > > 
> > > Atleast I got the query port assigned automatically. But I have no clue about how do I assign rcon port.
> > 
> > 
> > there is only 1 variable which is autoassigned: SERVER_PORT.
> > For all others you must add a varibale, so i dont believe that Queryport was autoassigned, when there is no variable
> 
> I guess I made it unclear what I did mean by that. Asssigning screenshot. I added server properties to the config file.
> 
> image

sry, but this is a hardcoded Port and this will never be added here

**Comment by Hendalf-DEV** [comment]
> > > > Hello again, I found the issue. It seems that POT does not want to run on Ubuntu image for some reason. So I reccomend change it to Debian image. That should work.
> > > > With kind regards, Hendalf.
> > > > p.s. I am going to make changes to this egg so no one will get to this trap again.
> > > 
> > > 
> > > Somehow I can't believe that now? Do you have an error message why it doesn't work with Ubuntu?
> > 
> > 
> > Nope, it just gives the same thing as this person had, I changed to Debian and now it works.
> 
> thats not a problem with the Distro. Which image are you using now?
> 
> If steamcmd, then Dotnet is allready installed. In Ubuntu it isnt

Ok, that makes sense. I use the ghcr.io/parkervcp/yolks:debian

**Comment by Hendalf-DEV** [comment]
> > > > > > The only thing that is additionally required is some of the ports If there is possibility to make them accessable for the server without adding them additionally, it would be nice.
> > > > > > image
> > > > > 
> > > > > 
> > > > > you can only autoassign the primary port. For all others you must add variables
> > > > 
> > > > 
> > > > Atleast I got the query port assigned automatically. But I have no clue about how do I assign rcon port.
> > > 
> > > 
> > > there is only 1 variable which is autoassigned: SERVER_PORT.
> > > For all others you must add a varibale, so i dont believe that Queryport was autoassigned, when there is no variable
> > 
> > 
> > I guess I made it unclear what I did mean by that. Asssigning screenshot. I added server properties to the config file.
> > image
> 
> sry, but this is a hardcoded Port and this will never be added here

My english could be a little bit worse, but I did not understand what did you mean by hardcoded port and what will not be added?

**Comment by That411Guy** [comment]
If you want I have a fully working egg for this game that I made

**Comment by Hendalf-DEV** [comment]
> If you want I have a fully working egg for this game that I made

This egg does work, so no need, thank you.

**Review by Hendalf-DEV — COMMENTED** [review]
All required changes are made, now need to test only the connection, and thats all.

File: game_eggs/path_of_titans/egg-path-of-titans.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    "author": "danielivanvladimir.revin@gmail.com",
+    "description": null,
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:debian": "ghcr.io\/parkervcp\/yolks:debian"
+    },
+    "file_denylist": [],
+    "startup": ".\/AlderonGamesCmd-Linux --game path-of-titans --server true --beta-branch {{BETA_BRANCH}} --auth-token {{AG_AUTH_TOKEN}} --install-dir .\/server\/ && .\/server\/PathOfTitansServer.sh {{SERVER_MAP}}?listen?MaxPlayers={{SERVER_MAX_PLAYERS}} -nullRHI -ServerName={{SERVER_NAME}} -ServerGUID={{SERVER_GUID}} -BranchKey={{BETA_BRANCH}} -Database={{SERVER_DB}} -log",
+    "config": {
+        "files": "{\r\n    \"server.properties\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"server-ip\": \"0.0.0.0\",\r\n            \"enable-query\": \"true\",\r\n            \"server-port\": \"{{server.build.default.port}}\",\r\n            \"query.port\": \"7778\"\r\n        }\r\n    }\r\n}",
```
Comment (gOOvER) [inline game_eggs/path_of_titans/egg-path-of-titans.json:18]: Port 7778 is hardcoded and this cant work, when the port is allready assigned in an other server. As i told you, add a variable for this port

**Review by gOOvER — CHANGES_REQUESTED** [review]

**Comment by Hendalf-DEV** [comment]
Server connectable
image

**Comment by Hendalf-DEV** [comment]
Stull awaiting review

**Comment by QuintenQVD0** [comment]
No pr will be merged to this repo as it soon will become read only. Please move your pr to the right repo on this organization as all eggs catogories now have their own repo.

**Comment by Hendalf-DEV** [comment]
> No pr will be merged to this repo as it soon will become read only. Please move your pr to the right repo on this organization as all eggs catogories now have their own repo.

Which repo is the right one?

**Comment by Hendalf-DEV** [comment]
Nevermind, I guess I found the one which is right one https://github.com/pelican-eggs/games. Atleast I hope it's the right one.

**Comment by QuintenQVD0** [comment]
> Nevermind, I guess I found the one which is right one https://github.com/pelican-eggs/games. Atleast I hope it's the right one.

That is indeed the right one for this PR

**Pull request closed without merging**